crates/uv-client/src/base_client.rs

```diff
@@ -148,19 +200,74 @@ impl<'a> BaseClientBuilder<'a> {
                 path_exists
             });
 
+            // Configure TLS.
+            let tls = {
+                // Set root certificates.
+                let mut root_cert_store = rustls::RootCertStore::empty();
+
+                if self.native_tls || ssl_cert_file_exists {
+                    let mut valid_count = 0;
+                    let mut invalid_count = 0;
+                    for cert in rustls_native_certs::load_native_certs().unwrap()
+                    {
+                        // Continue on parsing errors, as native stores often include ancient or syntactically
+                        // invalid certificates, like root certificates without any X509 extensions.
+                        // Inspiration: https://github.com/rustls/rustls/blob/633bf4ba9d9521a95f68766d04c22e2b01e68318/rustls/src/anchors.rs#L105-L112
+                        match root_cert_store.add(cert.into()) {
+                            Ok(_) => valid_count += 1,
+                            Err(err) => {
+                                invalid_count += 1;
+                                debug!("rustls failed to parse DER certificate: {err:?}");
+                            }
+                        }
+                    }
+                    if valid_count == 0 && invalid_count > 0 {
+                        // return Err(crate::error::builder(
+                        //     "zero valid certificates found in native root store",
+                        // ));
+                    }
+                } else{
+                    root_cert_store.extend(webpki_roots::TLS_SERVER_ROOTS.iter().cloned());
+                }
+
+                // Set TLS versions.
+                let versions = rustls::ALL_VERSIONS.to_vec();
+
+                // Allow user to have installed a runtime default.
+                // If not, we use ring.
+                let provider = rustls::crypto::CryptoProvider::get_default()
+                    .map(|arc| arc.clone())
+                    .unwrap();
+
+                let default_verifier = WebPkiServerVerifier::builder(Arc::new(root_cert_store)).build().unwrap();
+
+                // Build TLS config
+                let config_builder = ClientConfig::builder_with_provider(provider)
+                    .with_protocol_versions(&versions)
+                    .unwrap()
+                    .dangerous()
+                    .with_custom_certificate_verifier(Arc::new(CustomCertVerifier { hosts_to_skip_tls_verification: Default::default(), default_verifier }));
+
+                // Finalize TLS config
+                // STOPSHIP(charlie): Add `SSL_CERT` thing, it's private though? Ugh.
+                let mut tls = config_builder.with_no_client_auth();
+                tls.enable_sni = true;
```

Comment:
Basically stuck here, we need to read `SSL_CERT` and add it to `config_builder`, but the APIs that `reqwest` uses aren't public.